### DISCUSSION TITLE
how to open synth patterns in sampler, and viceversa?

### DISCUSSION THREAD

**Discussion by parrucasco** [bangcorrupt/hacktribe#336] [Q&A]
i already successfully installed hacktribe on my e2 (synth). but i cant seem to import patterns i made in normal firmware. is this possible? my question is also about .e2sallpat/.e2allpat and sample filles (.all)

**Comment by bangcorrupt** [comment]
Does it work with patterns if you change the file extension?

The sample files need modifying to use different slots if switching from sampler to Hacktribe.

**Reply by parrucasco** [reply]
already tried with changing extension from ".e2allpat" to ".e2sallpat" and it showed "File Error".

i am aware of changing sample slots, it's not that much work if i could load my old patterns in hacktribe :/

**Reply by bangcorrupt** [reply]
It looks like the synth patterns have the string `electribe` in the header, where the sampler patterns have `e2sampler`.  If you edit the headers so they match it should work.

**Reply by parrucasco** [reply]
sorry my ignorance...how to edit the header?

**Reply by bangcorrupt** [reply]
I added a [script](https://github.com/bangcorrupt/hacktribe/blob/main/scripts/e2pat_convert.py) to do this.  I only tested it briefly, but I was able to convert and import an allpat from the synth to Hacktribe.  It should work the same for allpat or individual patterns.

    python e2pat_convert.py input.e2pat sampler

This should create `converted-input.e2spat`.

**Reply by parrucasco** [reply]
thanks! how to execute this script? should i open a codespace inside hacktribe ? 
i will try this on the weekend
